src/main/scala/no/ndla/imageapi/service/search/IndexService.scala

```diff
@@ -110,16 +111,66 @@ trait IndexService {
       if (!indexWithNameExists(newIndexName).getOrElse(false)) {
         Failure(new IllegalArgumentException(s"No such index: $newIndexName"))
       } else {
-        oldIndexName match {
+        val actions = oldIndexName match {
           case None =>
-            e4sClient.execute(addAlias(ImageApiProperties.SearchIndex).on(newIndexName))
+            List[AliasActionDefinition](addAlias(ImageApiProperties.SearchIndex).on(newIndexName))
           case Some(oldIndex) =>
+            List[AliasActionDefinition](removeAlias(ImageApiProperties.SearchIndex).on(oldIndex), addAlias(ImageApiProperties.SearchIndex).on(newIndexName))
+        }
+
+        e4sClient.execute(aliases(actions)) match {
+          case Success(_) =>
+            logger.info("Alias target updated successfully, deleting other indexes.")
+            cleanupIndexes()
+          case Failure(ex) =>
+            logger.error("Could not update alias target.")
+            Failure(ex)
+        }
+      }
+    }
+
+    /**
+      * Deletes every index that is not in use by this indexService.
+      * Only indexes starting with indexName are deleted.
+      *
+      * @param indexName Start of index names that is deleted if not aliased.
+      * @return Name of aliasTarget.
+      */
+    def cleanupIndexes(indexName: String = ImageApiProperties.SearchIndex): Try[String] = {
+      e4sClient.execute(getAliases()) match {
+        case Success(s) =>
+          val indexes = s.result.mappings.filter(_._1.name.startsWith(indexName))
+          val unreferencedIndexes = indexes.filter(_._2.isEmpty).map(_._1.name).toList
```

Comment:
Kunne vært lurt å bruke `filter { case (x, y) => ...}` og navngi parameterene for å gjøre det litt tydeligere hva som skjer her :)